### PR TITLE
add ytdl

### PR DESCRIPTION
**Pull request by iiiiii1wepfj**

### CODE REVIEW

**Comment by alissonlauffer** [comment]
Why not using UserLixo ytdl?

**Comment by iiiiii1wepfj** [comment]
idk

**Comment by iiiiii1wepfj** [comment]
@alissonlauffer this is good?

**Comment by HitaloSama** [comment]
For me it's fine, but I think the `pretty_size` could be put in the `utils.py` and imported in `yt.py` since it may be useful for new plug-ins in the future and moving it into future commits would be unnecessary diff.

**Comment by HitaloSama** [comment]
> @HitaloSama From .yt import pretty_size?

Move the `pretty_size` to the `utils.py` and import it into the `yt.py`

```py
from utils import pretty_size
```

**Review by alissonlauffer — APPROVED** [review]
It is looking good to me now. Once @AndrielFR review the PR, it can be merged.

**Review by HitaloSama — APPROVED** [review]

File: plugins/yt.py [hunk 1]
```diff
@@ -1,105 +1,178 @@
...
     )
+    if not rege:
+        yt = await extract_info(ydl, "ytsearch:" + url, download=False)
+        yt = yt["entries"][0]
+    else:
+        yt = await extract_info(ydl, rege.group(), download=False)
+    for f in yt["formats"]:
+        if f["format_id"] == "140":
+            afsize = f["filesize"] or 0
+        if f["ext"] == "mp4" and not f["filesize"] is None:
```
Comment (AndrielFR) [inline plugins/yt.py]: It is not correct: `not var is None`, the correct thing is: `var is not None`.

File: plugins/yt.py [hunk 2]
```diff
@@ -1,105 +1,178 @@
...
+        yt = await extract_info(ydl, url, download=True)
+    except BaseException as e:
+        await cq.message.edit(strings("ytdlerrmsg").format(errmsg=e))
+        return
+    a = "Sending..."
+    await cq.message.edit(a)
+    filename = ydl.prepare_filename(yt)
+    ctime = time.time()
+    r = await http.get(yt["thumbnail"])
+    async with async_files.FileIO(f"{path}/{ctime}.png", "wb") as f:
```
Comment (AndrielFR) [inline plugins/yt.py]: The use of `io.BytesIO` is welcome.

**Review by AndrielFR — APPROVED** [review]
Despite the previous comments, everything seems to be ok.

File: plugins/yt.py (hunk 2 above)
Comment (alissonlauffer) [reply plugins/yt.py]: Agree, but this will not generate any blocks?

Comment (alissonlauffer) [reply plugins/yt.py]: ah, nvm. it is just a png file.

Comment (AndrielFR) [reply plugins/yt.py]: Yes. 😂

**Pull request merged**